From e2d6aaa1a809c0d49d599812440777549fc36ea4 Mon Sep 17 00:00:00 2001
From: Severin Gehwolf <sgehwolf@redhat.com>
Date: Thu, 22 Jul 2021 18:30:22 +0200
Subject: [PATCH] Backport JDK-8271148

---
 make/common/NativeCompilation.gmk | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/make/common/NativeCompilation.gmk b/make/common/NativeCompilation.gmk
index 490a7f4c7fb..2e3b5dd5eca 100644
--- a/make/common/NativeCompilation.gmk
+++ b/make/common/NativeCompilation.gmk
@@ -544,15 +544,6 @@ define SetupNativeCompilationBody
     # jmods.
     $1_OBJECT_DIR := $$($1_OBJECT_DIR)/static
     $1_OUTPUT_DIR := $$($1_OBJECT_DIR)
-    # For release builds where debug symbols are configured to be moved to
-    # separate debuginfo files, disable debug symbols for static libs instead.
-    # We don't currently support this configuration and we don't want symbol
-    # information in release builds unless explicitly asked to provide it.
-    ifeq ($(DEBUG_LEVEL), release)
-      ifeq ($(COPY_DEBUG_SYMBOLS), true)
-        $1_COMPILE_WITH_DEBUG_SYMBOLS := false
-      endif
-    endif
   endif
 
   ifeq ($$($1_TYPE), EXECUTABLE)
-- 
2.31.1

